libcst/metadata/scope_provider.py

```diff
@@ -401,10 +418,23 @@ def __init__(self, parent: "Scope") -> None:
         self._assignment_count = 0
 
     def record_assignment(self, name: str, node: cst.CSTNode) -> None:
-        self._assignments[name].add(
+        self._add_assignment(
             Assignment(name=name, scope=self, node=node, index=self._assignment_count)
```

Comment:
Since scope is set here, the old behavior of overriding `record_assignment` no longer changes which scope is recorded in the Assignment instance

Reply:
Thanks Luke, this was very frustrating!